src/tree/v2/azure/GroupingItem.ts

```diff
@@ -46,6 +48,14 @@ export class GroupingItem implements ResourceGroupsItem {
 
     async getChildren(): Promise<ResourceGroupsItem[] | undefined> {
         const sortedResources = this.resources.sort((a, b) => a.name.localeCompare(b.name));
+
+        this.onChangeBranchDataProviders((type: AzExtResourceType) => {
+            const azExtResourceTypes: AzExtResourceType[] = sortedResources.map(r => r.resourceType) as AzExtResourceType[];
```

Comment:
Why was this `as` cast necessary?

Reply:
I removed the cast, but now the type is `(AzExtResourceType | undefined)[]`, but I think that's fine.